### PR TITLE
[APM-4112] Add Benefit PG Source

### PR DESCRIPTION
**Pull request by andrew-berestov-cko**
# Description of changes in PR

[//]: Added Benefit PG Source

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [ ] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

**Review by alex-larin-cko — APPROVED** [review]

**Comment by andrew-berestov-cko** [comment]
@adam-young-cko Could you kindly approve this PR?

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Comment by sebastian-garcia-cardona-cko** [comment]
Please confirm updates with @Joshua-Bedeau-cko or Enrica Mameli before merging.

File: nas_spec/changelog.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 | Date       | Description of change                                                                                                |
 |------------|----------------------------------------------------------------------------------------------------------------------|
+| 2022/06/30 | Add Bebefit PG Request and Response sources                                                                                         |
```
Comment (Joshua-Bedeau-cko) [inline nas_spec/changelog.md:5]: Spelling mistake

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseBenefitSource.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+description: 'Benefit Source'
+allOf:
+  -
+    $ref: '#/components/schemas/PaymentResponseSource'
+  -
+    type: object
+    properties:
+      error_code:
+        type: string
+        description: 'The Benefit error code for transaction processing.'
```
Comment (Joshua-Bedeau-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseBenefitSource.yaml:14]: Have we agreed to put this in the source response? Even if we are not reusing the GW error code for this (which is another debate) an error code feels very much like processing to me and could change with each transaction for a given source. Can this please be flagged with APP Product.

**Review by Joshua-Bedeau-cko — CHANGES_REQUESTED** [review]

File: nas_spec/changelog.md (hunk 1 above)
Comment (andrew-berestov-cko) [reply nas_spec/changelog.md]: Thanks for noticing, Joshua. I have fixed that

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseBenefitSource.yaml (hunk 2 above)
Comment (andrew-berestov-cko) [reply nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseBenefitSource.yaml:14]: That one seems generic to all APMs. I think we can remove that. Will have a talk to anastasiya.piatrkouskaya

**Comment by tania-asimionesei-cko** [comment]
Please note that on the 27th of July we had a BREAKING CHANGE release, where the ABC and NAS versions of the API reference were switched in line with company-wide NAS readiness OKR and initiative. Therefore please make sure that you address any conflicts and updates in line with the latest release. 
PR for the NAS switch release can be found here: https://github.com/checkout/checkout-api-reference/pull/653

**Review by natasha-moore-cko — APPROVED** [review]

**Review by natasha-moore-cko — APPROVED** [review]

**Review by Enrica-Mameli-cko — APPROVED** [review]

**Review by natasha-moore-cko — APPROVED** [review]

**Pull request merged**